saleor/graphql/product/resolvers.py

```diff
@@ -60,24 +60,20 @@ def resolve_digital_contents(_info):
     return models.DigitalContent.objects.all()
 
 
-def resolve_product_by_id(info, id, channel_slug, requestor):
+def resolve_product_by_id_slug_or_ext_ref(
```

Comment:
Maybe just `resolve_product`? I would also change `ext_ref` to `external_reference` for readability.